src/client/pmix_client_get.c

```diff
@@ -626,6 +626,28 @@ static void _getnb_cbfunc(struct pmix_peer_t *pr,
                     PMIX_RELEASE(kv);
                 }
             }
+            else {
+               /* if we are both using the "hash" component, then the server's peer
+                * will simply be pointing at the same hash tables as my peer - no
+                * no point in checking there again */
+                if (0 != strcmp(pmix_globals.mypeer->nptr->compat.gds->name, pmix_client_globals.myserver->nptr->compat.gds->name)) {
+                    pmix_output_verbose(2, pmix_client_globals.get_output,
+                                        "pmix: get_nb searching for key %s for proc %s, - %s",
+                                        cb->key, PMIX_NAME_PRINT(&proc), pmix_client_globals.myserver->nptr->compat.gds->name);
+                    PMIX_GDS_FETCH_KV(rc, pmix_client_globals.myserver, cb);
+                }
```

Comment:
Argh - I should have caught this earlier. The "if success" clause below needs to be inside of the above "if" clause. Otherwise, you are checking `rc` again for no reason.

Reply:
Needs to go back to master branch as well - sorry for missing it earlier 😞

Reply:
I think it is correct in master - this is a bad cherry-pick. thanks, will fix.

Reply:
thx!